pkg/monitor/monitorapi/identification_pod.go

```diff
@@ -29,9 +29,9 @@ func NonUniquePodLocatorFrom(locator string) string {
 func PodFrom(locator string) PodReference {
 	parts := LocatorParts(locator)
 	namespace := NamespaceFrom(parts)
-	name := parts["pod"]
-	uid := parts["uid"]
-	if len(namespace) == 0 || len(name) == 0 || len(uid) == 0 {
+	name := parts[string(LocatorPodKey)]
+	uid := parts[string(LocatorUIDKey)]
+	if len(namespace) == 0 || len(name) == 0 {
```

Comment:
uid is allowed to be empty?

Reply:
> uid is allowed to be empty?

Yeah, the events being sent from the apiservers lack it for instance, but you still want to associate with pods.  I'd be happy to see the apiserver code updated to include uid.